op-e2e/system_test.go

```diff
@@ -1371,3 +1373,43 @@ func latestBlock(t *testing.T, client *ethclient.Client) uint64 {
 	require.Nil(t, err, "Error getting latest block")
 	return blockAfter
 }
+
+// TestPendingBlockIsLatest tests that we serve the latest block as pending block
+func TestPendingBlockIsLatest(t *testing.T) {
+	InitParallel(t)
+
+	cfg := DefaultSystemConfig(t)
+	sys, err := cfg.Start()
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+	l2Seq := sys.Clients["sequencer"]
+
+	t.Run("block", func(t *testing.T) {
+		for i := 0; i < 10; i++ {
+			pending, err := l2Seq.BlockByNumber(context.Background(), big.NewInt(-1))
+			require.NoError(t, err)
+			latest, err := l2Seq.BlockByNumber(context.Background(), nil)
+			require.NoError(t, err)
+			if pending.NumberU64() == latest.NumberU64() {
```

Comment:
Wouldn't this pass if the sequencer completed building the block between when we requested the pending block and requested the latest block?

Feels like these tests would be better in `op_geth_test` so they're controlling geth to avoid these kinds of race conditions.  Otherwise if we break this functionality in the future the test will fail intermittently (probably most of the time to be fair) rather than being guaranteed to fail.

Reply:
Added op-geth test